semantic-python/src/Language/Python/Core.hs

```diff
@@ -281,17 +281,17 @@ instance Compile Py.List
 instance Compile Py.ListComprehension
 
 instance Compile Py.Module where
-  compile it@Py.Module { Py.extraChildren = stmts } _cc = do
+  compile it@Py.Module { Py.extraChildren = stmts } _cc =
     -- This action gets passed to compile, which means it is the
     -- final action taken after the compiling fold finishes. It takes
     -- care of listening for the current set of bound variables (which
     -- is augmented by assignments and function definitions) and
     -- creating a record corresponding to those bindings.
-    let buildRecord = do
+    let buildRecord _ = do
           bindings <- asks @Bindings (toList . unBindings)
           let buildName n = (n, pure n)
           pure . record . fmap buildName $ bindings
-    foldr compile buildRecord stmts >>= locate it
+    in foldr compile buildRecord stmts >=> locate it
```

Comment:
I’m not 100% sure that `buildRecord` should be ignoring its argument, but again, this is probably one to understand via the tests.